k8s/solr/README.md

```diff
@@ -0,0 +1,417 @@
+environment by default.
+
+* [gcloud](https://cloud.google.com/sdk/gcloud/)
+* [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)
+* [docker](https://docs.docker.com/install/)
+* [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [openssl](https://www.openssl.org/)
+* [helm](https://helm.sh/)
+
+Configure `gcloud` as a Docker credential helper:
+
+```shell
+gcloud auth configure-docker
+```
+
+#### Creating a Google Kubernetes Engine (GKE) cluster
+
+Create a new cluster from the command line:
+
+```shell
+export CLUSTER=solr-cluster
+export ZONE=us-west1-a
+export PROJECT_ID=<GCP_Project_ID>
+
+gcloud container clusters create "$CLUSTER" --zone "$ZONE"
+```
+
+Configure `kubectl` to connect to the new cluster:
+
+```shell
+gcloud container clusters get-credentials "$CLUSTER" --zone "$ZONE"
+```
+
+#### Cloning this repo
+
+Clone this repo, and the associated tools repo:
+
+```shell
+git clone --recursive https://github.com/GoogleCloudPlatform/click-to-deploy.git
+```
+
+#### Installing the Application resource definition
+
+An Application resource is a collection of individual Kubernetes components,
+such as Services, StatefulSets, and so on, that you can manage as a group.
+
+To set up your cluster to understand Application resources, run the following
+command:
+
+```shell
+kubectl apply -f "https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/crd/app-crd.yaml"
+```
+
+You need to run this command once.
+
+The
+[Kubernetes SIG-apps](https://github.com/kubernetes/community/tree/master/sig-apps)
+community defines the Application resource. You can find the source code at
+[github.com/kubernetes-sigs/application](https://github.com/kubernetes-sigs/application).
+
+### Installing the app
+
+Navigate to the `solr` directory:
+
+```shell
+cd click-to-deploy/k8s/solr
+```
+
+#### Configuring the app with environment variables
+
+Choose the instance name and namespace for the app. For most cases, you can
+use the `default` namespace.
+
+```shell
+export APP_INSTANCE_NAME=solr-1
+export NAMESPACE=default
+```
+(Optional) Enable Stackdriver Metrics Exporter:
+
+> **NOTE:** Your GCP project must have Stackdriver enabled. If you are using a
+> non-GCP cluster, you cannot export metrics to Stackdriver.
+
+By default, the application does not export metrics to Stackdriver. To enable
+this option, change the value to `true`.
+
+```shell
+export METRICS_EXPORTER_ENABLED=false
+```
+
+Configure the image tag:
+
+```shell
+export TAG=8.6
+export ZK_TAG=3.6
+```
+Configure the container images:
+
+```shell
+export IMAGE_SOLR="marketplace.gcr.io/google/solr8"
+export IMAGE_ZOOKEEPER="marketplace.gcr.io/google/zookeeper3"
+export IMAGE_DEPLOYER="marketplace.gcr.io/google/solr/deployer:${TAG}"
+```
+
+Set or generate the passwords:
+
+```shell
+# SolrCloud UI console
+export SOLR_USER="solr"
+export SOLR_PASSWORD=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1 | tr -d '\n')
```

Comment:
I would use an alias for not repeating the three times the same command. Check here: https://github.com/GoogleCloudPlatform/click-to-deploy/blame/master/k8s/gitlab/README.md#L185